packages/peregrine/lib/talons/CheckoutPage/useCheckoutPage.js

```diff
@@ -56,31 +52,33 @@ export const useCheckoutPage = props => {
 
     const [
         getCheckoutDetails,
-        { data: checkoutData, loading: checkoutLoading }
-    ] = useLazyQuery(getCheckoutDetailsQuery, {
-        // TODO: Purposely overfetch and hit the network until all components
-        // are correctly updating the cache. Will be fixed by PWA-321.
-        fetchPolicy: 'cache-and-network'
-    });
+        {
+            data: checkoutData,
+            called: checkoutCalled,
+            client,
+            loading: checkoutLoading
+        }
+    ] = useLazyQuery(getCheckoutDetailsQuery);
 
-    const [getCheckoutStep, { data: stepData, client }] = useLazyQuery(
-        getCheckoutStepQuery
-    );
+    const checkoutStep = checkoutData && checkoutData.cart.checkoutStep;
 
     const setCheckoutStep = useCallback(
         step => {
+            const { cart: previousCart } = client.readQuery({
+                query: getCheckoutDetailsQuery
+            });
+
             client.writeQuery({
-                query: getCheckoutStepQuery,
+                query: getCheckoutDetailsQuery,
                 data: {
                     cart: {
-                        __typename: 'Cart',
-                        id: cartId,
+                        ...previousCart,
```

Comment:
Without the spread, would this blow away the entire Cart in the cache?

Reply:
I believe so but I can double check

Reply:
Without the spread the write still applies the new step but results in some warnings because the query provided expects certain fields. We can either pull off the fields from the `previousCart` and apply them or just spread the entire thing.

[![Image from Gyazo](https://i.gyazo.com/d6ef8a5514790d4f16fc541d7688e32f.png)](https://gyazo.com/d6ef8a5514790d4f16fc541d7688e32f)